samples/bluetooth/broadcast_audio_source/src/main.c

```diff
@@ -35,19 +35,80 @@ BUILD_ASSERT(CONFIG_BT_ISO_TX_BUF_COUNT >= TOTAL_BUF_NEEDED,
 #if defined(CONFIG_BAP_BROADCAST_16_2_1)
 
 static struct bt_bap_lc3_preset preset_active = BT_BAP_LC3_BROADCAST_PRESET_16_2_1(
-	BT_AUDIO_LOCATION_FRONT_LEFT, BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+	BT_AUDIO_LOCATION_FRONT_LEFT | BT_AUDIO_LOCATION_FRONT_RIGHT,
+	BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+
+#define BROADCAST_SAMPLE_RATE	16000
 
 #elif defined(CONFIG_BAP_BROADCAST_24_2_1)
 
 static struct bt_bap_lc3_preset preset_active = BT_BAP_LC3_BROADCAST_PRESET_24_2_1(
-	BT_AUDIO_LOCATION_FRONT_LEFT, BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+	BT_AUDIO_LOCATION_FRONT_LEFT | BT_AUDIO_LOCATION_FRONT_RIGHT,
+	BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+
+#define BROADCAST_SAMPLE_RATE	24000
 
 #endif
 
+#if defined(CONFIG_LIBLC3)
+#include "lc3.h"
+
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+#include <zephyr/usb/usb_device.h>
+#include <zephyr/usb/class/usb_audio.h>
+#include <zephyr/sys/ring_buffer.h>
+
+/* USB Audio Data is downsampled from 48kHz stereo to 16kHz mono when receiving data */
+#define USB_SAMPLE_RATE         48000
+#define USB_DOWNSAMPLE_RATE     BROADCAST_SAMPLE_RATE
+#define USB_FRAME_DURATION_US   1000
+#define USB_NUM_SAMPLES         ((USB_FRAME_DURATION_US * USB_DOWNSAMPLE_RATE) / USEC_PER_SEC)
+#define USB_BYTES_PER_SAMPLE    2
+#define USB_CHANNELS		2
+
+#define AUDIO_RING_BUF_BYTES    USB_NUM_SAMPLES * USB_BYTES_PER_SAMPLE * 20
+#else /* !defined(CONFIG_USB_DEVICE_AUDIO) */
+
+#include <math.h>
+
+#define AUDIO_VOLUME            (INT16_MAX - 3000) /* codec does clipping above INT16_MAX - 3000 */
+#define AUDIO_TONE_FREQUENCY_HZ   400
+
+/**
+ * Use the math lib to generate a sine-wave using 16 bit samples into a buffer.
+ *
+ * @param buf Destination buffer
+ * @param length_us Length of the buffer in microseconds
+ * @param frequency_hz frequency in Hz
+ * @param sample_rate_hz sample-rate in Hz.
+ */
+static void fill_audio_buf_sin(int16_t *buf, int length_us, int frequency_hz, int sample_rate_hz)
+{
+	const int sine_period_samples = sample_rate_hz / frequency_hz;
+	const unsigned int num_samples = (length_us * sample_rate_hz) / USEC_PER_SEC;
+	const float step = 2 * 3.1415f / sine_period_samples;
+
+	for (unsigned int i = 0; i < num_samples; i++) {
+		const float sample = sin(i * step);
+
+		buf[i] = (int16_t)(AUDIO_VOLUME * sample);
+	}
+}
+#endif /* defined(CONFIG_USB_DEVICE_AUDIO) */
+#endif /* defined(CONFIG_LIBLC3) */
+
 static struct broadcast_source_stream {
 	struct bt_bap_stream stream;
 	uint16_t seq_num;
 	size_t sent_cnt;
+#if defined(CONFIG_LIBLC3)
+	lc3_encoder_t lc3_encoder;
+	lc3_encoder_mem_16k_t lc3_encoder_mem;
```

Comment:
`lc3_encoder_mem_16k_t` probably won't work for 24KHz

Reply:
good find - I'll try to get 24KHz working all the way (requires some test setup to receive also)

Reply:
saw this too late - last push adds fixes for 24khz (apparently, the next level is the 48khz type)